ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -213,23 +221,37 @@ private TransactionSelectionResult evaluateTransaction(
     final WorldUpdater worldStateUpdater = worldState.updater();
     final BlockHashLookup blockHashLookup = new BlockHashLookup(processableBlockHeader, blockchain);
 
-    final TransactionProcessingResult result =
-        transactionProcessor.processTransaction(
-            blockchain,
-            worldStateUpdater,
-            processableBlockHeader,
-            transaction,
-            miningBeneficiary,
-            blockHashLookup,
-            false,
-            TransactionValidationParams.mining());
-
-    if (!result.isInvalid()) {
+    TransactionProcessingResult effectiveResult;
+    if (transaction.isGoQuorumPrivateTransaction()) {
+
+      final ValidationResult<TransactionInvalidReason> validationResult =
+          validateTransaction(processableBlockHeader, transaction, worldStateUpdater);
+      if (!validationResult.isValid()) {
+        return TransactionSelectionResult.CONTINUE;
+      }
```

Comment:
I think we do not need these three lines, as this is checked in 249 and following!

Reply:
I think we do still need the return here

Reply:
refactored